### PR TITLE
Added new value to return output of proofpoint TAP plugin

### PR DESCRIPTION
**Pull request by tylerslijboom-r7**
### Description

- Add a new return key and value to the Proofpoint TAP action.  The Threat Details URL is a link to Proofpoint documentation on the threat in element "threat_details_url"

## Testing

Some fields obfuscated with <removed>

```
run -R tests/parse_tap_alert_attachment.json 
INFO[0000] Running command:  docker run --rm -i rapid7/proofpoint_tap:1.0.5 run < tests/parse_tap_alert_attachment.json 
{"body": {"log": "rapid7/Proofpoint TAP:1.0.5. Step name: parse_tap_alert", "status": "ok", "meta": {}, "output": {"results": {"threat": {"attachment_sha256": "", "category": "phish", "condemnation_time": "2019-08-12T22:03:36Z", "url": "<removed>", "threat_details_url": "https://threatinsight.proofpoint.com/<removed>/threat/email/<removed>"}, "message": {"time_delivered": "2019-08-10T05:18:51Z", "recipients": "<removed>", "subject": "97", "sender": "<removed>", "header_from": "97", "header_replyto": "97", "message_id": "<removed>", "sender_ip": "103.27.74.25", "message_size": "97"}, "browser": {"time": "2019-08-12T14:11:47Z", "source_ip": “<removed>“, "user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.140 Safari/537.36 Edge/17.17134"}}}}, "version": "v1", "type": "action_event"}

```

```
icon-plugin run -R tests/parse_tap_alert_attachment_bad_input.json
INFO[0000] Running command:  docker run --rm -i rapid7/proofpoint_tap:1.0.5 run < tests/parse_tap_alert_attachment_bad_input.json 
{"body": {"log": "rapid7/Proofpoint TAP:1.0.5. Step name: parse_tap_alert\n", "status": "ok", "meta": {}, "output": {"results": {"threat": {"attachment_sha256": "", "category": "phish", "condemnation_time": "2019-08-12T22:03:36Z", "url": "hxxp://tags[.]my/htaccess/file/index[.]php?e user@example.com", "threat_details_url": ""}, "message": {"time_delivered": "2019-08-10T05:18:51Z", "recipients": "user@example.com", "subject": "97", "sender": "user@example.com", "header_from": "97", "header_replyto": "97", "message_id": "user@example.com", "sender_ip": "103.27.74.25", "message_size": "97"}, "browser": {"time": "2019-08-12T14:11:47Z", "source_ip": "192.234.69.178", "user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) Ap pleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.140 Safari/537.36 Edg e/17.17134"}}}}, "version": "v1", "type": "action_event"}

```

## Input 

Screen Shot 2019-12-09 at 3 17 31 PM

Screen Shot 2019-12-09 at 3 17 56 PM

## Output

Screen Shot 2019-12-09 at 3 23 35 PM

## Artifacts

Screen Shot 2019-12-09 at 3 24 46 PM

### CODE REVIEW

**Comment by tylerslijboom-r7** [comment]
> @tylerslijboom-r7 missing PR assessment. Also, if the URL is not available in the input, will the action still succeed? Might be good to have a negative test.

added a test with bad input, _bad_input.json.  

Does not fail, just does not return a value.

File: proofpoint_tap/bin/komand_proofpoint_tap [hunk 1]
```diff
@@ -5,8 +5,8 @@ from komand_proofpoint_tap import connection, actions, triggers
 
 
 Name = 'Proofpoint TAP'
-Vendor = 'rapid7'
-Version = '1.0.4'
+Vendor = 'tslijboom'
```
Comment (gusingh-r7) [inline proofpoint_tap/bin/komand_proofpoint_tap]:
```suggestion
Vendor = 'rapid7'
```

File: proofpoint_tap/help.md [hunk 2]
```diff
@@ -78,6 +79,7 @@ This plugin does not contain any troubleshooting information.
 
 # Version History
 
+* 1.0.5 - Parsing out the "View Threat Details" link from emails to its own value
```
Comment (gusingh-r7) [inline proofpoint_tap/help.md]: We may not need double quotes here.

File: proofpoint_tap/plugin.spec.yaml [hunk 3]
```diff
@@ -4,8 +4,8 @@ products: [insightconnect]
 name: proofpoint_tap
 title: Proofpoint TAP
 description: A plugin for Proofpoint Targeted Attack Protection (TAP) alerts
-version: 1.0.4
-vendor: rapid7
+version: 1.0.5
+vendor: tslijboom
```
Comment (gusingh-r7) [inline proofpoint_tap/plugin.spec.yaml]:
```suggestion
vendor: rapid7
```

File: proofpoint_tap/setup.py [hunk 4]
```diff
@@ -2,10 +2,10 @@
 from setuptools import setup, find_packages
 
 
-setup(name='proofpoint_tap-rapid7-plugin',
-      version='1.0.4',
+setup(name='proofpoint_tap-tslijboom-plugin',
```
Comment (gusingh-r7) [inline proofpoint_tap/setup.py]:
```suggestion
setup(name='proofpoint_tap-rapid7-plugin',
```

File: proofpoint_tap/setup.py [hunk 5]
```diff
@@ -2,10 +2,10 @@
...
+      version='1.0.5',
       description='A plugin for Proofpoint Targeted Attack Protection (TAP) alerts',
-      author='rapid7',
+      author='tslijboom',
```
Comment (gusingh-r7) [inline proofpoint_tap/setup.py]:
```suggestion
      author='rapid7',
```

File: proofpoint_tap/plugin.spec.yaml [hunk 6]
```diff
@@ -42,6 +42,11 @@ types:
       description: Condemnation Time
       type: string
       required: false
+    threat_details_url:
+      title: Threat Details URL
+      description: URL for Details of the Threat
```
Comment (jschipp-r7) [inline proofpoint_tap/plugin.spec.yaml]:
```suggestion
      description: URL for details of the threat
```

File: proofpoint_tap/plugin.spec.yaml [hunk 7]
```diff
@@ -4,7 +4,7 @@ products: [insightconnect]
 name: proofpoint_tap
 title: Proofpoint TAP
 description: A plugin for Proofpoint Targeted Attack Protection (TAP) alerts
-version: 1.0.4
+version: 1.0.5
```
Comment (jschipp-r7) [inline proofpoint_tap/plugin.spec.yaml]: This should be a minor version bump per https://semver.org which is what we use.

**Pull request merged**